metrics/web/integration_config.yml

```diff
@@ -0,0 +1,15 @@
+source:
```

Comment:
It's probably out of scope for this - but as we can use CI Integration tool in Github Actions directly it might be easier just to pass all needed parameters as arg parameter to the tool rather than storing config files somewhere?

so it might be something like
```
./ci_integrations sync --config-string "
source:
  github_actions:
    workflow_identifier: web_actions.yml
    repository_name: monorepo
    repository_owner: platform-platform
.....
"
```
so we can mix-in env variables

Some issues that I see with the config file approach:
1. We are saving secrets there before starting sync - file can be accidentally exported as an artifact
2. It feels more natural to store those things in GitHub config ".github" folder as they are project specific. We already have some config in index.html but I think we would want to move it out as well.
3. Files are very similar - so it probably doesn't make sense to duplicate them for every integration - maybe we should think of an approach to reuse shared configuration items between integrations inside monorepo?

Reply:
I think config in yaml is a common pattern. I'm not sure we'd want to pass complex YAML as a CLI arg.

Reply:
Flank uses yaml extensively for config: 
- https://github.com/Flank/flank/blob/master/test_runner/flank.yml
- https://github.com/Flank/flank/blob/master/test_projects/android/flank.yml

Reply:
We can add the `config` command to the CI integrations CLI that would help a user to manage the configuration file's content. For example, let's configure the GitHub Actions config YAML: 
```bash 
ci_integrations config github create --access_token=$ACCESS_TOKEN --repository_owner=$OWNER # A user can fill the missing fields interactively using CLI 
ci_integrations config github show # or it could be: list, print, etc. 
ci_integrations config github clear # or it could be: delete, remove, etc. 
ci_integrations config github edit # interactively edits config file 
```
Then, the `ci_integrations sync` command will use the default config file created with the `config` command if exists. Otherwise, will look for a file by the given path (current `--config-file` option). Finally, if all of the above fails - the `sync` command fails as the config is missing.

Or we can provide an environment for the `ci_integrations sync` command and use it to replace variables within the given config file. For example, if the given file contains the `TOKEN` variable: 
```bash 
ci_integrations sync --config-file=/path/to/config.yaml --env TOKEN=$ACCESS_TOKEN 
```

As the third option, we can investigate how [aqueduct](https://pub.dev/packages/aqueduct) deals with env variables in config files since they work properly. 

And finally, I wonder if it is possible to add the whole config content to the secrets and then just `echo` this content to the `config.yaml`? 
```bash
echo $SECRET_CONFIG >> /path/to/config.yaml
ci_integrations sync --config-file=/path/to/config.yaml
```

Hope this comment is useful, thanks! 🙂

Reply:
Thanks for your input @solid-vladlaba ! I think we can save this for a future discussion and for now just go with yaml files as was previously discussed and how it's implemented right now.